contracts/src/periphery/EigenDACertVerifierV2.sol

```diff
@@ -0,0 +1,218 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.9;
+
+import {IEigenDACertVerifierV2} from "src/interfaces/IEigenDACertVerifierV2.sol";
+import {IEigenDAThresholdRegistry} from "src/interfaces/IEigenDAThresholdRegistry.sol";
+import {IEigenDABatchMetadataStorage} from "src/interfaces/IEigenDABatchMetadataStorage.sol";
+import {IEigenDASignatureVerifier} from "src/interfaces/IEigenDASignatureVerifier.sol";
+import {EigenDACertVerificationV1Lib as CertV1Lib} from "src/libraries/EigenDACertVerificationV1Lib.sol";
+import {EigenDACertVerificationV2Lib as CertV2Lib} from "src/libraries/EigenDACertVerificationV2Lib.sol";
+import {OperatorStateRetriever} from "lib/eigenlayer-middleware/src/OperatorStateRetriever.sol";
+import {IRegistryCoordinator} from "lib/eigenlayer-middleware/src/RegistryCoordinator.sol";
+import {IEigenDARelayRegistry} from "src/interfaces/IEigenDARelayRegistry.sol";
+import {
+    BatchHeaderV2,
+    BlobInclusionInfo,
+    NonSignerStakesAndSignature,
+    SignedBatch,
+    SecurityThresholds
+} from "src/interfaces/IEigenDAStructs.sol";
+
+/**
+ * @title A CertVerifier is an immutable contract that is used by a consumer to verify EigenDA blob certificates
+ * @notice For V2 verification this contract is deployed with immutable security thresholds and required quorum numbers,
+ *         to change these values or verification behavior a new CertVerifier must be deployed
+ */
+contract EigenDACertVerifierV2 is IEigenDACertVerifierV2 {
+    error InvalidSecurityThresholds();
+
+    /// @notice The EigenDAThresholdRegistry contract address
+    IEigenDAThresholdRegistry public immutable eigenDAThresholdRegistryV2;
```

Comment:
function names need the version suffix for backwards compatibility, but do these fields? Can't this be named like e.g. `eigenDAThresholdRegistry`?

Reply:
Only if it were private! It's probably not ideal, but this contract is meant to be inherited to mirror the current interface of the CertVerifier, which means that if any other inherited contract declared the same variables, the compiler would not accept this. I plan to delete the CertVerifier in a future PR, which would allow us to remove this annoyance, but this PR prioritizes replicating the previous CertVerifier behavior by splitting up V1 and V2 and relying on inheritance.